src/table_builder.cc

```diff
@@ -115,16 +106,24 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
                         index.file_number, get_status.ToString().c_str());
       }
     }
-    if (builder_unbuffered()) {
-      base_builder_->Add(key, value);
-    } else {
-      std::unique_ptr<BlobFileBuilder::BlobRecordContext> ctx =
-          NewCachedRecordContext(ikey, value);
-      blob_builder_->AddSmall(std::move(ctx));
-    }
+    AddSmallToTableAdaptively(ikey, value);
   } else {
-    assert(builder_unbuffered());
-    base_builder_->Add(key, value);
+    // Mainly processing kTypeMerge and kTypeBlobIndex in both flushing and compaction.
+    AddSmallToTableAdaptively(ikey, value);
+  }
+}
+
+void TitanTableBuilder::AddSmallToTableAdaptively(const ParsedInternalKey& ikey, const Slice& value){
```

Comment:
Rename it to `AddToBaseTable`, and change L59 and L67 too

Reply:
If so, there will be two AddToBaseTable functions in this class. How about AddSmallToBaseTable?

Reply:
Well, another AddToBaseTable is also adding small values.
I will rename it with AddToBaseTable.

Reply:
emmm... let's call it `AddSmall`